cmd/jb/install.go

```diff
@@ -58,18 +59,23 @@ func installCommand(dir, jsonnetHome string, uris []string) int {
 
 			// we want to install the passed version (ignore the lock)
 			delete(lockFile.Dependencies, d.Name)
+
+			// we will need to write jsonnetfiles
+			depsChanged = true
 		}
 	}
 
 	locked, err := pkg.Ensure(jsonnetFile, jsonnetHome, lockFile.Dependencies)
 	kingpin.FatalIfError(err, "failed to install packages")
 
-	kingpin.FatalIfError(
-		writeJSONFile(filepath.Join(dir, jsonnetfile.File), jsonnetFile),
-		"updating jsonnetfile.json")
-	kingpin.FatalIfError(
-		writeJSONFile(filepath.Join(dir, jsonnetfile.LockFile), spec.JsonnetFile{Dependencies: locked}),
-		"updating jsonnetfile.lock.json")
+	if depsChanged {
```

Comment:
I feel this is too naive, especially because it could miss cases where we need to write although we have not passed any URIs.

For example when running `jb install` without any arguments, this would prevent it from writing the lockfile.

A more robust implementation would be to use a sth like the following:
```go
func checkJsonnetfile(path string, jf *jsonnetFile) bool {
  b := open(path)
  orig := unmarshal(b)

  return deepEqual(orig, jf)
}

Wouldn't say that this is overengineered, rather just a robust solution. Probably better to check twice here than not enough, as it could break a lot of things
```

Reply:
And check both files separately :D